client/errors.go

```diff
@@ -1,9 +1,100 @@
 package client
 
-import "errors"
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
 
 var (
```

Comment:
Another way of ensuring backward compatibility is:
`ErrNotFound = problems.NotFound`
since it implements error
I might even be tempted to mark these as deprecated and hint to use problems instead

The only thing that needs to be backwards compatible in theory is the exposed API, i.e. the errors

Reply:
> If we all agree that we want to support go errors + problems. This is great :)

Not all APIs can be changed to use Problems, therefore either we create two Client implementations or support both. I think maintaining two separate client libraries might be a lot of work. The purpose of this library is to be a base of all our go-clients, so if we don't put the logic here we need to place it in the different service clients instead.

> ErrNotFound = problems.NotFound

I think this would be a dangerous change. The problem fields will be set and we can not ensure that all APIs return the same problem details (as specified in OpenAPI) as we define here.

> I might even be tempted to mark these as deprecated and hint to use problems instead

This is quite hard to mark since it requires changes not in the client but in the servers. I would like everyone to use Problems but due to our API contracts, this is not a trivial change.